asdfghjkl/precondition.py

```diff
@@ -47,82 +46,94 @@ def _get_fisher(self, module, postfix=None):
+                                       data_average=data_average,
+                                       seed=seed,
+                                       scale=scale,
                                        n_mc_samples=self.n_mc_samples)
         self.fisher_manager = rst
 
-    def move_curvature(self, postfix, scale=1., ema_decay=None, to_pre_inv=False):
-        self.accumulate_curvature(postfix, scale, ema_decay, to_pre_inv, replace=True)
-
-    def accumulate_curvature(self, postfix='acc', scale=1., ema_decay=None, to_pre_inv=False, replace=False):
-        if ema_decay is None:
-            ema_decay = self.ema_decay
-        if to_pre_inv:
-            postfix = self._pre_inv_postfix
-        for module in self.modules:
-            fisher = self._get_fisher(module)
-            if fisher is None:
-                continue
-            fisher.scaling(scale)
-            fisher_acc = self._get_fisher(module, postfix)
-            if fisher_acc is None or replace or ema_decay == 1.0:
-                self._set_fisher(module, fisher, postfix)
-            else:
-                fisher.scaling(ema_decay)
-                fisher_acc.scaling(1.0 - ema_decay)
-                fisher_acc += fisher
-            self._clear_fisher(module)
-
-    def finalize_accumulation(self, postfix='acc'):
-        for module in self.modules:
-            fisher_acc = self._get_fisher(module, postfix)
-            assert fisher_acc is not None
-            self._set_fisher(module, fisher_acc)
-            self._clear_fisher(module, postfix)
+    def accumulate_curvature(self,
+                             inputs=None,
+                             targets=None,
+                             data_loader=None,
+                             ema_decay=None,
+                             data_average=True,
+                             seed=None,
+                             scale=1):
+        self._update_curvature(inputs,
+                               targets,
+                               data_loader,
+                               accumulate=True,
+                               ema_decay=ema_decay,
+                               data_average=data_average,
+                               seed=seed,
+                               scale=scale)
+
+    def refresh_curvature(self,
+                          inputs=None,
+                          targets=None,
+                          data_loader=None,
+                          data_average=True,
+                          seed=None,
+                          scale=1):
+        if self.ema_decay is not None:
+            warnings.warn(f'ema_decay ({self.ema_decay}) will be ignored.')
+        self._update_curvature(inputs,
+                               targets,
+                               data_loader,
+                               accumulate=False,
+                               data_average=data_average,
+                               seed=seed,
+                               scale=scale)
```

Comment:
I think this `_update_curvature` leads to assertion error on line 66 if `self.ema_decay is not None`.

Reply:
Yes, @ishiihiro is right. Although `refresh_curvature()` ignores `self.ema_decay`, `_update_curvature()` tries to refer it anyway. I will fix this part.